### PR TITLE
Papaboard

### PR DESCRIPTION
**Pull request by Chrisc110**
Created a new can message type for the radiation board (papaboard), build function, get function, and unit test.

---
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/canlib/32)

### CODE REVIEW

File: message_types.h [hunk 1]
```diff
@@ -91,10 +95,12 @@
  * GPS_INFO:        TSTAMP_MS_H TSTAMP_MS_M  TSTAMP_MS_L    NUM_SAT                 QUALITY         None            None            None
  *
  * FILL_LVL:        TSTAMP_MS_H TSTAMP_MS_M  TSTAMP_MS_L    FILL_LEVEL              DIRECTION       None            None            None
- *
+ * 
+ * RADI_VALUE:      TSTAMP_MS_H TSTAMP_MS_M  TSTAMP_MS_L    RADI_BOARD              RADI_INT        RADI_DECI
```
Comment (akmorrison) [inline message_types.h:99]: You should be explicit here about what bytes 6 and 7 are meant to be, and if they're unused write out `None`. Also, if you are splitting RADI_DECI across two bytes, please write one of them as RADI_DECI_H (the higher order bits) and one as RADI_DECI_L (the lower order bits)

File: message_types.h [hunk 2]
```diff
@@ -159,6 +165,7 @@ enum SENSOR_ID {
     SENSOR_PRESSURE_CC,
     SENSOR_VENT_BATT,
     SENSOR_INJ_BATT,
+    SENSOR_RADIATION,
```
Comment (akmorrison) [inline message_types.h:168]: Does this get used anywhere? This enum is mostly for `SENSOR_ANALOG` messages I think, if you aren't using this enum anywhere you probably don't need to add this line

File: tests/unit/build_can_message.c [hunk 3]
```diff
@@ -576,6 +576,60 @@ bool test_build_fill_message(void)
     return ret;
 }
 
+bool test_build_radi_info_msg (void)
+{
+    uint32_t timestamp = 0x123456;
+    can_msg_t output;
+    uint8_t sensor_identifier = 3;
+    uint8_t int_value = 3;
+    uint16_t deci_value = 1;
```
Comment (akmorrison) [inline tests/unit/build_can_message.c:585]: If `deci_value` is meant to be two bytes long (i.e. have a max value of 65535) you should test with a higher value than this, just to make sure that all the bits are copied in/out properly.

File: message_types.h [hunk 4]
```diff
@@ -91,10 +95,12 @@
...
  *
  * FILL_LVL:        TSTAMP_MS_H TSTAMP_MS_M  TSTAMP_MS_L    FILL_LEVEL              DIRECTION       None            None            None
- *
+ * 
+ * RADI_VALUE:      TSTAMP_MS_H TSTAMP_MS_M  TSTAMP_MS_L    RADI_BOARD              RADI_INT        RADI_DECI
+ * 
  * LEDS_ON:         None        None         None           None                    None            None            None            None
  * LEDS_OFF:        None        None         None           None                    None            None            None            None
- *
+ *
```
Comment (akmorrison) [inline message_types.h:103]: I'd recommend running a formatting script over your changes before committing, it'll keep the codebase consistent and stop any trailing whitespace from getting in.

[Here's](https://waterloorocketry.slack.com/archives/C01535M46SC/p1613883526003200) a link to the astylerc (config file for the [astyle](http://astyle.sourceforge.net/)) tool that I used and that shouldn't (I hope) change anything else in the files except what you've added.

**Comment by Chrisc110** [comment]
All the suggestions have been resolved!

**Review by zx100x100 — APPROVED** [review]
Looks good, before you merge please uprev the revision number on line 71 of message_types.h and line 12 of can.h

**Review by zx100x100 — APPROVED** [review]
Looks good :partyparrot:

**Pull request merged**